src/stream/src/task/stream_manager.rs

```diff
@@ -251,26 +216,62 @@ impl LocalStreamManager {
             .await
     }
 
-    pub async fn clear_storage_buffer(&self) {
-        dispatch_state_store!(self.state_store.clone(), store, {
-            store.clear_shared_buffer().await.unwrap();
-        });
+    pub fn context(&self) -> &Arc<SharedContext> {
+        &self.context
+    }
+
+    /// Drop the resources of the given actors.
+    pub async fn drop_actors(&self, actors: Vec<ActorId>) -> StreamResult<()> {
+        self.local_barrier_manager
+            .send_and_await(|result_sender| LocalBarrierEvent::DropActors {
+                actors,
+                result_sender,
+            })
+            .await
+    }
+
+    /// Force stop all actors on this worker, and then drop their resources.
+    pub async fn reset(&self) {
+        self.local_barrier_manager
+            .send_and_await(LocalBarrierEvent::Reset)
+            .await
+            .expect("should receive reset")
+    }
+
+    pub async fn update_actors(&self, actors: Vec<stream_plan::StreamActor>) -> StreamResult<()> {
+        self.local_barrier_manager
+            .send_and_await(|result_sender| LocalBarrierEvent::UpdateActors {
+                actors,
+                result_sender,
+            })
+            .await?
+    }
+
+    /// This function could only be called once during the lifecycle of `LocalStreamManager` for
```

Comment:
Is there some way we can enforce that this function is only called once at most?

Reply:
I think the doc string is outdated. I will removed it.